File: app/src/test/java/org/simon/npo/cucumber/NpoTestContext.java
package org.simon.npo.cucumber;

import java.util.HashMap;
import java.util.Map;

public class NpoTestContext {
  private final Map<String, Object> map = new HashMap<>();
  private static final String API_RESPONSE_KEY = "API_RESPONSE";

  public void putApiResponse(Object voidResponseEntity) {
    map.put(API_RESPONSE_KEY, voidResponseEntity);
  }

  public Object getApiResponse() {
    return map.get(API_RESPONSE_KEY);
  }
}

File: app/src/test/kotlin/org/simon/npo/cucumber/NpoTestContext.kt
package org.simon.npo.cucumber

import org.springframework.http.ResponseEntity

@Suppress("UNCHECKED_CAST")
class NpoTestContext(
    private val map: MutableMap<String, Any> = HashMap()) {
    fun putApiResponse(voidResponseEntity: Any) {
        map[API_RESPONSE_KEY] = voidResponseEntity
    }

    fun <T> getApiResponse(): ResponseEntity<T> = map[API_RESPONSE_KEY] as ResponseEntity<T>
}

const val  API_RESPONSE_KEY = "API_RESPONSE"
